platforms/jvm/toolchains-jvm-shared/build.gradle.kts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.distribution.api-java")
+}
+
+description = "Declarations to define JVM toolchains shared between launcher and daemon"
+
+errorprone {
+    disabledChecks.addAll(
+        "StringCaseLocaleUsage", // 2 occurrences
+        "UnnecessaryLambda", // 2 occurrences
+    )
+}
+
+dependencies {
+    implementation(project(":core"))
```

Comment:
Absolutely agree with this, however, unfortunately isn't possible to use the `FileSystemOperations` since we also need `zipTree` and `tarTree` that are not available. Definitely, a refactor will be great to extract those from the `core` module and expose them in some util module avoiding in this way to depend on the huge monolith. 

Having said that let's keep this refactor separate and create a dedicated ticket for it.